### PR TITLE
[Danger] Add some static check rules in dangerfile

### PR DESCRIPTION
**Pull request by wrmswindmill**
# Brief Description of the PR
This PR add the following features:
1.  Check if PR is submitted to master branch,if not ,the Travis CI will build fail.
2. Check if PR is binded with a milestone, if not, The Travis CI will output a warning message.
3. Check if your PR description contains keywords `Documentation ` and it's correponding http links.if not, The Travis CI will output a warning message.
4. Check if your PR description contains keywords `Demo` and it's correponding http links.if not, The Travis CI will output a warning message.
5. update the  changelog check rule message

# Checklist
* [] Demo
* [ ] Documentation:https://github.com/apache/incubator-weex-site/pull/447

### CODE REVIEW

File: dangerfile.js [hunk 1]
```diff
@@ -16,15 +16,67 @@
...
+}
+
+var pr_body = danger.github.pr.body.toLowerCase();
+// Because Pr description template include the following line：
+// 1. Write the corresponding [documentation](https://weex.io/guide/contribute/how-to-contribute.html#contribute-code-or-document)
+// so we should check the documentation below the ### checklist
+console.log("checkDocumentation");
+const index = pr_body.indexOf("checklist")
+const includeChecklist = (index!=-1)
+if(includeChecklist && !matchRegex(pr_body.substring(index),/documentation.*http/)){
```
Comment (YorkShen) [inline dangerfile.js]: You should allow both of the following spelling:
* documentation
* Documentation

File: dangerfile.js [hunk 2]
```diff
@@ -16,15 +16,67 @@
...
+  const msg = "if you update the code, "+
+    "maybe you should update the documentation and add the documentation link in the PR description. \n" +
+    "here is the guide about how to contribute documentation:https://weex.apache.org/guide/contribute/how-to-contribute.html#contribute-code-or-document \n";
+  warn(msg);
+}
+
+// check if pr contains a demo link
+console.log("checkDemo");
+if(!matchRegex(pr_body,/demo.*http/)){
```
Comment (YorkShen) [inline dangerfile.js]: You should allow both of the following spelling:
* demo
* Demo

File: dangerfile.js [hunk 3]
```diff
@@ -16,15 +16,67 @@
...
 import fs from "fs";
 import path from 'path';
 import GitHubApi from 'github';
 import parseDiff from 'parse-diff';
 
+// check if pr submitted to master branch
+console.log("checkMasterBranch")
+const isMergeRefMaster = danger.github.pr.base.ref === 'master';
+if(!isMergeRefMaster){
+  fail("you must submit PR to master branch");
```
Comment (YorkShen) [inline dangerfile.js]: * Change it to warning.
* Change the message to the following
    > You'd better to submit PR to master branch as the development of Weex is on master branch.

File: dangerfile.js [hunk 4]
```diff
@@ -16,15 +16,67 @@
...
+}
+
 // Make sure there are changelog entries
 const hasChangelog = danger.git.modified_files.includes("changelog.md")
-if (!hasChangelog) { warn("No Changelog changes!") }
+if (!hasChangelog) { 
+  const title = "No Changelog changes!"
+  const idea =  "Can you add a Changelog?"+
+    "To do so,append your changes to the changelog.md "
+  warn(`${title} - <i>${idea}</i>`);
```
Comment (YorkShen) [inline dangerfile.js]: FYI: You can use `markdown` function in danger

File: dangerfile.js [hunk 5]
```diff
@@ -16,15 +16,67 @@
...
+
+var pr_body = danger.github.pr.body.toLowerCase();
+// Because Pr description template include the following line：
+// 1. Write the corresponding [documentation](https://weex.io/guide/contribute/how-to-contribute.html#contribute-code-or-document)
+// so we should check the documentation below the ### checklist
+console.log("checkDocumentation");
+const index = pr_body.indexOf("checklist")
+const includeChecklist = (index!=-1)
+if(includeChecklist && !matchRegex(pr_body.substring(index),/documentation.*http/)){
+  const msg = "if you update the code, "+
```
Comment (YorkShen) [inline dangerfile.js]: The first letter should be in upper case, e.g. **If**

File: dangerfile.js [hunk 6]
```diff
@@ -16,15 +16,67 @@
...
+  const msg =  "if your PR is about fixing a bug excluding crash the code,"+
+    "you should add the demo link in the PR description. \n "+
+    "here is a demo link:http://dotwe.org/vue?spm=a2c7j.-guide-contribute-contribute-code.0.0.3e93748cmxz3yt";
+  warn(msg);
+}
+
+// check if pr bind the github milestone
+console.log("checkMileStone");
+if(!danger.github.pr.milestone){
+  warn("current pr not bind the milestone");
```
Comment (YorkShen) [inline dangerfile.js]: The first letter should be in upper case, e.g. **Current**.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -6,13 +6,16 @@ All PRs should be submitted to master branch -->
...
 
 
 # Brief Description of the PR
 
 # Checklist
-* [] Demo
-* [] Document
+<!-- For example,if you write the corresponding Documentation,you should list it below with the format:
+[x] Documentation:http-links -->
+* [] Demo:
```
Comment (YorkShen) [inline .github/PULL_REQUEST_TEMPLATE.md]: The following markdown would be enough:
```
   * Demo:
   * Documentation
```

File: dangerfile.js (hunk 1 above)
Comment (wrmswindmill) [reply dangerfile.js]: I hava consider this situation.

```
var pr_body = danger.github.pr.body.toLowerCase();
```

File: dangerfile.js (hunk 2 above)
Comment (wrmswindmill) [reply dangerfile.js]: this is the same as above

File: dangerfile.js (hunk 5 above)
Comment (wrmswindmill) [reply dangerfile.js]: I have updated the code .

File: dangerfile.js (hunk 6 above)
Comment (wrmswindmill) [reply dangerfile.js]: I have updated the code .

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (wrmswindmill) [reply .github/PULL_REQUEST_TEMPLATE.md]: I have updated the code .

File: dangerfile.js (hunk 4 above)
Comment (wrmswindmill) [reply dangerfile.js]: I'm sorry, I don't understand it.

File: dangerfile.js (hunk 5 above)
Comment (wrmswindmill) [reply dangerfile.js]: I hava updated the code

File: dangerfile.js (hunk 4 above)
Comment (wrmswindmill) [reply dangerfile.js]: use `markdown` function is more readable. I have updated the code

**Comment by YorkShen** [comment]
The check rules are described in https://github.com/apache/incubator-weex/blob/master/CONTRIBUTING.md#publish-your-change

**Pull request merged**